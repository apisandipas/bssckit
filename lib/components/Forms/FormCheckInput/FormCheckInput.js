"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledTools = require("styled-tools");

var _theme = _interopRequireDefault(require("../../../theme"));

var _base = require("../../Utilities/base");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  margin-top: ", ";\n  margin-left: ", ";\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n    & + label {\n      color: ", ";\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var getColor = function getColor(props) {
  return props.disabled && (0, _styledComponents.css)(_templateObject(), (0, _styledTools.theme)('formCheckInput.colors.default.colorDisabled'), (0, _styledTools.theme)('formCheckInput.colors.default.colorDisabledLabel'));
};

var FormCheckInput = (0, _styledComponents["default"])(_base.Input)(_templateObject2(), function (props) {
  return (0, _styledTools.theme)('formCheckInput.margin.top');
}, function (props) {
  return (0, _styledTools.theme)('formCheckInput.margin.left');
}, function (props) {
  return getColor(props);
});
FormCheckInput.defaultProps = {
  theme: _theme["default"]
};
var _default = FormCheckInput;
exports["default"] = _default;