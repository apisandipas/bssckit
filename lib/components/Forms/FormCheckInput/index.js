"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "FormCheckInput", {
  enumerable: true,
  get: function get() {
    return _FormCheckInput["default"];
  }
});

var _FormCheckInput = _interopRequireDefault(require("./FormCheckInput"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }