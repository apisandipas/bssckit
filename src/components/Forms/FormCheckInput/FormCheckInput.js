import styled, { css } from 'styled-components';
import defaultTheme from '../../../theme';
import { Input } from '../../Utilities/base';
import { getColor, getMargin } from '../../../utils/themeFunctions';

const color = props =>
  props.disabled &&
  css`
    color: ${getColor(props, 'formCheckInput', 'colorDisabled')};
    & + label {
      color: ${getColor(props, 'formCheckInput', 'colorDisabledLabel')};
    }
  `;

const FormCheckInput = styled(Input)`
  margin-top: ${props => getMargin(props, 'formCheckInput', 'top')};
  margin-left: ${props => getMargin(props, 'formCheckInput', 'left')};
  ${props => color(props)};
`;

FormCheckInput.defaultProps = {
  theme: defaultTheme,
};

export default FormCheckInput;
