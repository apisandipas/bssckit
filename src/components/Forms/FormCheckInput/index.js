export { default as FormCheckInput } from './FormCheckInput';
