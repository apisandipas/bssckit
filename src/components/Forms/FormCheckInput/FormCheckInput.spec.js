import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import FormCheckInput from './FormCheckInput';

describe('FormInline Component', () => {
  it('matches normal FormCheckInput', () => {
    const { container } = render(<FormCheckInput />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches FormCheckInput disabled', () => {
    const { container } = render(<FormCheckInput disabled />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
