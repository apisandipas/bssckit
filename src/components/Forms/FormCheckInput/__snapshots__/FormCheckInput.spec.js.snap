// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInline Component matches FormCheckInput disabled 1`] = `
.c0 {
  margin-top: 0.3rem;
  margin-left: 0rem;
  color: #abaeb1;
}

.c0 + label {
  color: #abaeb1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<input
  class="sc-fzoyAV c0"
  disabled=""
/>
`;

exports[`FormInline Component matches normal FormCheckInput 1`] = `
.c0 {
  margin-top: 0.3rem;
  margin-left: 0rem;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<input
  class="sc-fzoyAV c0"
/>
`;
